data/oot/world/fire_temple.yml

```diff
@@ -3,28 +3,28 @@
   exits:
     "Death Mountain Crater Warp": "true"
     "Fire Temple Lava Room": "has_small_keys_fire(1)"
-    "Fire Temple Boss": "has(BOSS_KEY_FIRE) && (event(FIRE_TEMPLE_PILLAR_HAMMER) || has_hover_boots) && has_tunic_goron_strict"
+    "Fire Temple Boss": "has(BOSS_KEY_FIRE) && (event(FIRE_TEMPLE_PILLAR_HAMMER) || has_hover_boots) && has_tunic_goron"
     "Fire Temple Boss Key Loop": "cond(setting(smallKeyShuffle, anywhere), has(SMALL_KEY_FIRE, 8), true) && can_hammer"
   locations:
-    "Fire Temple Jail 1 Chest": "true"
+    "Fire Temple Jail 1 Chest": "has_tunic_goron"
 "Fire Temple Boss Key Loop":
   dungeon: Fire
   locations:
-    "Fire Temple Boss Key Side Chest": "can_hammer"
-    "Fire Temple Boss Key Chest": "can_hammer"
-    "Fire Temple GS Hammer Statues": "can_hammer"
+    "Fire Temple Boss Key Side Chest": "true"
+    "Fire Temple Boss Key Chest": "true"
+    "Fire Temple GS Hammer Statues": "true"
 "Fire Temple Lava Room":
   dungeon: Fire
   exits:
-    "Fire Temple After Elevator": "has_small_keys_fire(3) && has(STRENGTH) && (has_ranged_weapon_adult || has_explosives) && has_tunic_goron_strict"
+    "Fire Temple After Elevator": "has_small_keys_fire(3) && has_tunic_goron_strict"
```

Comment:
You need to push the big block and trigger the switch for the ring of fire

Reply:
Keep reading. I changed the Fire Temple a lot. This is covered.